test/functional/feature_backwards_compatibility.py

```diff
@@ -156,173 +150,72 @@ def run_test(self):
+                    assert info['keypoolsize'] > 0
+                    txs = wallet.listtransactions()
+                    assert_equal(len(txs), 5)
+                    assert_equal(txs[1]["txid"], tx1_id)
+                    assert_equal(txs[2]["walletconflicts"], [tx1_id])
+                    assert_equal(txs[1]["replaced_by_txid"], tx2_id)
+                    assert not(txs[1]["abandoned"])
+                    assert_equal(txs[1]["confirmations"], -1)
+                    assert_equal(txs[2]["blockindex"], 1)
+                    assert txs[3]["abandoned"]
+                    assert_equal(txs[4]["walletconflicts"], [tx3_id])
+                    assert_equal(txs[3]["replaced_by_txid"], tx4_id)
+                    assert not(hasattr(txs[3], "blockindex"))
+                elif wallet_name == "w2":
+                    assert(info['private_keys_enabled'] == False)
+                    assert info['keypoolsize'] == 0
+                else:
+                    assert(info['private_keys_enabled'] == True)
+                    assert info['keypoolsize'] == 0
         else:
-            # Descriptor wallets appear to be corrupted wallets to old software
-            assert_raises_rpc_error(-4, "Wallet file verification failed: wallet.dat corrupt, salvage failed", node_v19.loadwallet, "w1")
-            assert_raises_rpc_error(-4, "Wallet file verification failed: wallet.dat corrupt, salvage failed", node_v19.loadwallet, "w2")
-            assert_raises_rpc_error(-4, "Wallet file verification failed: wallet.dat corrupt, salvage failed", node_v19.loadwallet, "w3")
-            assert_raises_rpc_error(-4, "Wallet file verification failed: wallet.dat corrupt, salvage failed", node_v18.loadwallet, "w1")
-            assert_raises_rpc_error(-4, "Wallet file verification failed: wallet.dat corrupt, salvage failed", node_v18.loadwallet, "w2")
-            assert_raises_rpc_error(-4, "Wallet file verification failed: wallet.dat corrupt, salvage failed", node_v18.loadwallet, "w3")
-
-        # Open the wallets in v0.17
-        node_v17.loadwallet("w1_v18")
-        wallet = node_v17.get_wallet_rpc("w1_v18")
-        info = wallet.getwalletinfo()
-        assert info['private_keys_enabled']
-        assert info['keypoolsize'] > 0
-
-        node_v17.loadwallet("w2_v18")
-        wallet = node_v17.get_wallet_rpc("w2_v18")
-        info = wallet.getwalletinfo()
-        assert info['private_keys_enabled'] == False
-        assert info['keypoolsize'] == 0
+            for node in self.nodes[2:-1]:
+                # Descriptor wallets appear to be corrupted wallets to old software
+                # and loadwallet is introduced in v0.17.0
+                if node.version >= 170000 and node.version < 210000:
+                    for wallet_name in ["w1", "w2", "w3"]:
+                        assert_raises_rpc_error(-4, "Wallet file verification failed: wallet.dat corrupt, salvage failed", node.loadwallet, wallet_name)
```

Comment:
In commit "test: v0.20.1 backwards compatibility" (94d3989352aba63c86c901269438bfd566dd8632)

Looking at all the test changes here it seems like no test coverage is lost, but it would be nice if commit message could say whether this is the case.

IMO, it would also be nicer if this commit were split into a refactoring commit that did not change coverage and just deduped code, and a smaller followup commit adding the new v0.20 version. But just mentioning however this commit affects coverage would be good if you don't want to change the commit contents.

Reply:
I moved the refactoring stuff into the misc changes commit. Adding a note that perhaps coverage changed in the process of this cleanup.